src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/ProofCertificateDownload.swift

```diff
@@ -0,0 +1,104 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+public struct ProofCertificateDownload {
+
+    // MARK: - Public
+
+    public func fetchProofCertificate(
+        for healthCertificates: [Base45],
+        with httpService: HTTPServiceProtocol = HTTPService(),
+        completion: @escaping (Result<CBORData?, ProofCertificateFetchingError>
+    ) -> Void) {
+
+        let certificateAccess = DigitalGreenCertificateAccess()
+
+        let eligibleCertificates =
+            healthCertificates.compactMap { (base45) -> CBORData? in
+                let result = certificateAccess.extractCBOR(from: base45)
+                switch result {
+                case .success(let healthCertificateCBORData):
+                    return healthCertificateCBORData
+                case .failure:
+                    return nil
+                }
+            }.filter {
+                switch certificateAccess.extractDigitalGreenCertificate(from: $0) {
+                case .success(let certificate):
+                    return certificate.vaccinationCertificates[0].isEligibleForProofCertificate
+                case .failure:
+                    return false
+                }
+            }
+
+        guard !eligibleCertificates.isEmpty else {
+            // ToDo: At the moment this point is tbd in the spec.
+            completion(.success(nil))
+            return
+        }
+
+        // This call recursively posts the eligible health certificates one after the other.
+        // The operation completes, as soon as the first proof certificate is returned from the backend.
+        fetchProofCertificateRecursion(for: eligibleCertificates, with: httpService, completion: completion)
+    }
+
+    // MARK: - Private
+
+    private func fetchProofCertificateRecursion(
+        for healthCertificates: [CBORData],
+        with httpService: HTTPServiceProtocol = HTTPService(),
+        completion: @escaping (Result<CBORData?, ProofCertificateFetchingError>) -> Void
+    ) {
+
+        let url = URL(string: "https://api.certify.demo.ubirch.com/api/certify/v2/reissue/cbor")
```

Comment:
Should we handle this more flexible or are there no staging servers?

Reply:
Environment implementation will follow in an own PR